### ISSUE TITLE
New Issue discovered with EthMiner V15 with console mining to Minergate

### ISSUE THREAD

**Issue by RUminingTwo** [ethereum-mining/ethminer#1430]
I had thought the V15 of Ethminer was conquered with getting it to work with Minergate settings, mining in a console for ETC

Version 14 worked fine, and still works fine with the following settings, when using Minergate.

Works
ethminer -U -S etc.pool.minergate.com:45777 -u UserEmailAddress@EmailDomain

Version 15 the settings we thought would properly work as as follows.

ethminer -U stratum+tcp://UserEmailAddress%40EmailDomain@etc.pool.minergate.com:45777

The settings about seemed to initially work, but what i have found is that it is NOT reporting proper has rates, by far amounts, to where is should be MegaHash (MH), it reports KiloHash, (kH).

I actually believe it is only logging the worker, and no hashrate whatsoever.

Then I had tried Mingergate's, updates settings for Ver 15 of EthMIner 

Cuda and OpenCL
 Doesn't work ----  ethminer -U -S etc.pool.minergate.com:45777/UserEmailAddress@emaildomain
or
 Doesn't work ----  ethminer -G -S etc.pool.minergate.com:45777/UserEmailAddress@emaildomain

So to sum it up
Ver 14 Ethminer still works with 
ethminer -U -S etc.pool.minergate.com:45777 -u UserEmailAddress@EmailDomain

Version 15 works Sort of, Not reporting proper hashrate by far, if at all 
ethminer -U stratum+tcp://UserEmailAddress%40EmailDomain@etc.pool.minergate.com:45777

Minergates web site info / config to use with V15 of EthMiner

 Doesn't work ----  ethminer -U -S etc.pool.minergate.com:45777/UserEmailAddress@emaildomain
or
 Doesn't work -

Version 14 seems to still be the safe bet.

**Comment by AndreaLanfranchi** [comment]
>Minergate's updated web site info / config to use with V15 of EthMiner
Doesn't work ---- ethminer -U -S etc.pool.minergate.com:45777/UserEmailAddress@emaildomain
or
Doesn't work ---- ethminer -G -S etc.pool.minergate.com:45777/UserEmailAddress@emaildomain

If they (Minergate) have updated their site with these settings they are completely **wrong**

-S argument is no longer live and the syntax has no meaning.

Proper usage is

ethminer -G -P stratum+tcp://etc.pool.minergate.com:45777/UserEmailAddress@emaildomain

or 

ethminer -U -P stratum+tcp://etc.pool.minergate.com:45777/UserEmailAddress@emaildomain

**Comment by AndreaLanfranchi** [comment]
>The settings about seemed to initially work, but what i have found is that it is NOT reporting proper has rates, by far amounts, to where is should be MegaHash (MH), it reports KiloHash, (kH).

Please note that ethminer does not report neither MegaHashes per second nor KiloHashes per second.
We send out Hashes per second. **It's up to the pool to scale the received value to what they prefer** We do not perform any scaling.

**Comment by AndreaLanfranchi** [comment]
CC @MinerGate

**Comment by AndreaLanfranchi** [comment]
We're not responsible for what you see on minergate's console.

**Comment by RUminingTwo** [comment]
Yes, Minergate has updated their site to reflect Ver 15, 

https://minergate.com/altminers/ethminer

What MG had posted for the script, doesn't work, the settings you have recommend does work, but it does not show any info on Minergate's site for each GPU MH rate. Its only showing the hashrate for the CPU's I am also making use of.

Version 14 would report on the Console., and on MG's site the actual hash rate of MH rate properly on both. 

""We're not responsible for what you see on minergate's console.""""
Understood, I figured since they entered our conversation, and took note from it, in which the day after they had updated to V15. I figured they would be here to view again.

**Comment by AndreaLanfranchi** [comment]
They're also making mistake with the version of ethminer.

They point to http://ethdocs.org/en/latest/ethereum-clients/cpp-ethereum/installing-binaries/index.html#installing-binaries

Which has, nowadays, nothing to do with ethminer

**Comment by RUminingTwo** [comment]
I seen that also, I followed the link today, and was like Hugh? Their link doesn't even match up to the commands & info they are posting.

Little scary, And I think I am going to start looking at specific ETC pools, as I had transferred what I had mined out today, and they first concert it to ETC, and then transfer it out to the exchange I am using.

Any reputable ETC/ETH exchanges you could you could recommend, with a reasonable fee.

**Comment by AndreaLanfranchi** [comment]
Do you mean a "pool" ?
Exchanges are not viable for ethminer's connections

**Comment by RUminingTwo** [comment]
Yes, I meant pool, sorry.

**Comment by RUminingTwo** [comment]
In what I was mentioning before regarding EthMiner console reporting hash rate, here is a screen capture, ( hope it posts) and this is using Ver 14, and MGate is properlly reporting the MH rate the console is reporting.
![image](https://user-images.githubusercontent.com/35960173/43790709-63d78f10-9a41-11e8-8245-806df1b8f931.png)

**Comment by AndreaLanfranchi** [comment]
You can safely use ethermine.org or nanopool
Both have 1% fee

**Comment by AndreaLanfranchi** [comment]
> and this is using Ver 14, and MGate is properlly reporting the MH rate the console is reporting.

I'm pretty sure in console of ver 0.16 you see the same values.

**Comment by RUminingTwo** [comment]
THank you on the two pools, will check them out and test em.

I will also try V16, although that is still beta? no?

**Comment by RUminingTwo** [comment]
Will check 16 out once I can get past these few bumps, and the testing of an Alternate ETC pool

**Comment by RUminingTwo** [comment]
Testing out Ethermine.org, with Ver 15 
Console setup is 

ethminer -U -P stratum1+tcp://MyWalletAddress@us1-etc.ethermine.org:4444

While on the console it states 21.38 Mh/s, the Etherminer.org. shows only 7.6 MH/s?

Is there something I should be aware of or configured wrong?

Any feedback would be greatly appreciated.
Thanks!

**Comment by RUminingTwo** [comment]
Maybe they dont update their reporting on hashrate as fast, is what i am guessing

**Comment by AndreaLanfranchi** [comment]
What you see (I believe) is a bumpy red line which reports your **calculated** hashrate which is based on the number of submitted shares in the last 15 minutes (if I recall well).
Then there is a blue line which averages the red line on the last 2 hours.
None of the above matches your reported hashrate.

If you want to report hashrate to pool you have to explicitly tell ethminer to do so.
Thus your command line has to be

```
ethminer -U -P stratum1+tcp://MyWalletAddress@us1-etc.ethermine.org:4444 --report-hashrate
```
or ... using short syntax
```
ethminer -U -P stratum1+tcp://MyWalletAddress@us1-etc.ethermine.org:4444 -R
```

A good reading of the output from `ethminer --help` is your best friend.

**Comment by AndreaLanfranchi** [comment]
Probably you had the same problem on MG

**Comment by RUminingTwo** [comment]
Ah, that I wasn't aware of with regards to the  -- Report-hashrate. I will add that command line argument to the console batch file.

In resepec to Minergate, it has been prettty accurate over last 6 months or so, the only issue I had with reporting was when I dropped in the V15, but as we see they have not installed or configured to the newer version (V15) properlly or something.

Ethermine, is now finally reporting the accurate hash rate on their web, compared to the desktop console/command line.

Seems they update like you had mentioned, every 15 minutes, and now both match with the hash rate..

You had mentioned a good reading, on the help, yes, I have to utilize it more, and test more, One of the reasons I had used Minergate was to keep it simple for me, so I dont have to spend much time with all the config options, or such, but now I am becoming a little bit more familiar with the configs. Didn't take long to get the ethermine.org config going..

But still have some more read the options and testing those.

Now I can copy the batch files to the other systems, and try using all my systems with Ethermine.org instead of the MG for ETC.

Thanks again for your help!
Mucho Appreiciado
 : )
